src/apps/frontend/pages/tasks/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import TaskModal from './task-modal';
+import constant from '../../constants';
+import TaskHeader from './task-header';
+import { Task } from '../../types/task';
+import TaskList from './task-list';
+
+const Tasks: React.FC = () => {
+  const [tasks, setTasks] = useState<Task[]>([]);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const addTaskFormik = useFormik({
+    initialValues: {
+      id: '',
+      title: '',
+      description: '',
+    },
+    validationSchema: Yup.object({
+      title: Yup.string()
+        .min(constant.ADD_TASK_MIN_LENGTH, constant.ADD_TASK_VALIDATION_ERROR)
+        .required(constant.ADD_TASK_VALIDATION_ERROR),
+      description: Yup.string()
+        .min(
+          constant.ADD_TASK_DESCRIPTION_MIN_LENGTH,
+          constant.ADD_TASK_DESCRIPTION_VALIDATION_ERROR,
+        )
+        .required(constant.ADD_TASK_DESCRIPTION_VALIDATION_ERROR),
+    }),
+    onSubmit: (values, { resetForm }) => {
+      if (!values.title || !values.description) {
+        console.error('Task details are missing!');
+        return;
+      }
+      setTasks((prevTasks) => [
+        ...prevTasks,
+        { ...values, id: Date.now().toString() },
+      ]);
+      resetForm();
+      setIsModalOpen(false);
+    },
+  });
+
+  return (
+    <div className="p-4 mx-auto max-w-(--breakpoint-2xl) md:p-6">
+      <TaskHeader onAddTask={() => setIsModalOpen(true)} />
+      <div className="shadow-default rounded-2xl border border-[#e4e7ec]">
```

Comment:
Someone who did construct the markup - it is quite hard and time consuming to deconstruct what this, and div below are trying to do.

If you see in the video here, [How to build maintainable frontends that future dev's would thank you for](https://www.loom.com/share/c4352f7c0be949e8bc0377dbcba15daa), we generally recommend creating components with meaningful layout so the markup is easy to read.

For example, if you a row with two columns, ideal markup should be:

```
<Row>
  <Column></Column>
  <Column></Column>
</Row>
```

instead of div with flex css classes.

This practice helps to create more maintainable codebase.